### PR TITLE
resolved bug #11447 command-completion of :runtime should list files …

### PR DESCRIPTION
**Pull request by cero1988**
resolved bug #11447 command-completion of :runtime should list files in runtime path

### CODE REVIEW

**Comment by brammool** [comment]
```
	char *directories[] = {"", NULL};
```

The empty string is used to create a pattern for the matched path:

```
	sprintf((char *)s, "%s/%s*.vim", dirnames[i], pat);
```

Have a try using "**" instead of the empty string, doesn't that work
better?

However, if the pattern already includes a path separator, e.g.
"syntax/f", then it's probably better not to use a directory and only
append "*.vim".

-- 
hundred-and-one symptoms of being an internet addict:
33. You name your children Eudora, Mozilla and Dotcom.

 /// Bram Moolenaar -- ***@***.*** -- http://www.Moolenaar.net   \\\
///                                                                      \\\
\\\        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by cero1988** [comment]
you are right, it's working better with **

**Review by yegappan — COMMENTED** [review]
Can you add a test for this new completion type in test_cmdline.vim?

Also, you need to add "runtime" to command_complete[] in usercmd.c.

**Comment by cero1988** [comment]
> Can you add a test for this new completion type in test_cmdline.vim?
> 
> Also, you need to add "runtime" to command_complete[] in usercmd.c.

I have added the test in test_cmdline.vim and the runtime to command_complete

File: src/cmdexpand.c [hunk 1]
```diff
@@ -3019,6 +3026,12 @@ ExpandFromContext(
 	return ExpandRTDir(pat, DIP_START + DIP_OPT, numMatches, matches,
 								directories);
     }
+    if (xp->xp_context == EXPAND_RUNTIME)
+    {
+	char *directories[] = {"", NULL};
+	return ExpandRTDir(pat, DIP_START + DIP_OPT + DIP_PRNEXT, numMatches, matches,
+								directories);
```
Comment (yegappan) [inline src/cmdexpand.c:3033]: From the above codecov output, it looks like most of the key changes are not covered by tests. Can you modify the tests to cover these lines?

File: src/cmdexpand.c [hunk 2]
```diff
@@ -2311,6 +2313,11 @@ set_context_by_cmdname(
 	    xp->xp_context = EXPAND_COLORS;
 	    xp->xp_pattern = arg;
 	    break;
+	
+        case CMD_runtime:
+	    xp->xp_context = EXPAND_RUNTIME;
+            xp->xp_pattern = arg;
+	    break;
```
Comment (yegappan) [inline src/cmdexpand.c:2321]: Can you update the tests to cover the above lines?  You can look at the Test_cmdline_complete_various() function in the test_cmdline.vim file for some examples for this test.

**Pull request closed without merging**